### ISSUE TITLE
Error when !Sub contains a literal

### ISSUE THREAD

**Issue by KenHundley** [aws-cloudformation/cfn-python-lint#1435]
Version: cfn-lint 0.29.0

We've have some instances where we use string literals in !Sub functions (`${!...}`), but this is throwing an error in the linter.

Ex.
```yaml
authorizerUri: !Sub "arn:aws:apigateway:${AWS::Region}:lambda:path/2015-03-31/functions/arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:function:${!stageVariables.authorizerName}/invocations"
```
This example is throwing the error: `Unable to find referenced value, '!stageVariables'`

This functionality is defined on the page: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-sub.html

>To write a dollar sign and curly braces (${}) literally, add an exclamation point (!) after the open curly brace, such as ${!Literal}. AWS CloudFormation resolves this text as ${Literal}.

The same template when scanned with `aws cloudformation validate-template` does pass without error and it can be successfully deployed.

**Comment by KenHundley** [comment]
nm, I just realized that it wasn't the linter that was causing the issue.
